internal/controller/deployment_controller.go

```diff
@@ -87,7 +89,66 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return r.Update(ctx, l, deployment)
 }
 
+func (r *DeploymentReconciler) setStatusFromClusterStatus(ctx context.Context, dc *dynamic.DynamicClient, l logr.Logger, deployment *hmc.Deployment) (bool, error) {
+	resourceId := schema.GroupVersionResource{
+		Group:    "cluster.x-k8s.io",
+		Version:  "v1beta1",
+		Resource: "clusters",
+	}
+
+	list, err := dc.Resource(resourceId).Namespace(deployment.Namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		l.Info("Cluster not found, ignoring since object must be deleted or not yet created")
```

Comment:
This is a list object but the info returned implies a single cluster, perhaps `No clusters found` is a better log message here.